SharedUtils/src/main/java/com/braintreepayments/api/sharedutils/SynchronousHttpClient.kt

```diff
@@ -0,0 +1,66 @@
+package com.braintreepayments.api.sharedutils
+
+import androidx.annotation.RestrictTo
+import java.net.HttpURLConnection
+import javax.net.ssl.HttpsURLConnection
+import javax.net.ssl.SSLSocketFactory
+
+/**
+ * This class performs an http request on the calling thread. The external caller is
+ * responsible for thread scheduling to ensure that this is not called on the main thread.
+ */
+@RestrictTo(RestrictTo.Scope.LIBRARY_GROUP)
+internal class SynchronousHttpClient(
+    private val socketFactory: SSLSocketFactory,
+    private val parser: HttpResponseParser
+) {
+
+    @Throws(Exception::class)
+    fun request(httpRequest: HttpRequest): HttpResponse {
+        requireNotNull(httpRequest.path) { "Path cannot be null" }
+        val url = httpRequest.url
+
+        val startTime = System.currentTimeMillis()
```

Comment:
Not sure if this spacing makes more sense, take it or leave it:
```suggestion
        requireNotNull(httpRequest.path) { "Path cannot be null" }

        val url = httpRequest.url
        val startTime = System.currentTimeMillis()
```